modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/exp/agg/AccumulatorsFactory.java

```diff
@@ -214,6 +214,11 @@ private WrapperPrototype(AggregateCall call) {
             List<RelDataType> inTypes = SqlTypeUtil.projectTypes(inputRowType, call.getArgList());
             List<RelDataType> outTypes = accumulator.argumentTypes(ctx.getTypeFactory());
 
+            if (call.getArgList().size() > outTypes.size()) {
```

Comment:
Why not
```
assert call.getArgList().size() <= outTypes.size() : "Unexpected number of arguments: " +
                    "expected=" + outTypes.size() + ", actual=" + inTypes.size());
```

Reply:
Just to throw an exception with a meaningful message, rather than ArrayIndexOutOfBound.